src/extras/amd.js

```diff
@@ -135,6 +139,13 @@
   global.define.amd = {};
 
   function addToRegisterRegistry(name, define) {
+    if (!firstNamedDefine) {
+      firstNamedDefine = define;
+      setTimeout(function () {
+        firstNamedDefine = null;
+      });
```

Comment:
We shouldn't rely on a `setTimeout`. Rather we should clear this when we know it is safe to clear it. If this code path is the only knowledge that we should clear it then perhaps we should have a `shouldClearNamedDefine` variable.

Reply:
I thought through how to do this without setTimeout earlier - one approach I considered to was to reset `firstNamedDefine` inside of `getRegister()`, or perhaps inside of `instantiate`.

Could you explain how this `shouldClearNamedDefine` variable would work?